README.en

```diff
@@ -0,0 +1,357 @@
+    * operator + (const Fsm&): returns the concatenation of machines that allows
+      strings divided into two parts so that the first is allowed by the first machine,
+      and the second - by the second machine
+    * operator | (const Fsm&): returns the union of machines that allows
+      strings allowed by at least one machine
+    * operator * (): returns the machine iteration that allows
+      strings representing a repetition of a string allowed by the original machine
+      0 or more times (the so-called Kleene star)
+    * operator ~ (): returns the machine inversion that allows strings now allowed by the
+      original machine
+    * operator & (const Fsm&): returns the machine intersection that allows strings allowed by
+      both machines
+    * operator * (size_t n): returns the machine that allows strings representing a repetition of
+      a string allowed by the original machine n times
+    * Surrounded(): returns the machine that allows any strings containing the string allowed by the
+      original machine (adds /.*/ to the start or the end)
+    * operator +=, operator |=, Iterate, Complement, operator &=, operator *=, and Surround the same to
+      the seven described above
+    * Reverse(): returns the machine that allows strings that are a mirror of strings allowed by
+      the original machine
+
+Combined machine building is available, for example: (lexer1.Parse() | lexer2.Parse()).Compile<Scanner>()).
+For example, it allows to read a list of patterns from a file, compile them into Fsm, and combine them all
+into one large Fsm which can be compiled into a scanner.
+
+Scanners
+=======
+
+There are three existing scanners in Pire: Scanner, SimpleScanner, and SlowScanner.
+
+    * Scanner is the main engine of all Pire. This scanner was mentioned above.
+    * SimpleScanner is the simplified version of the scanner suitable for checking the text with one simple regular expression.
+      It does not support machines union, and its transition table is slightly larger than in Scanner,
+      but SimpleScanner works about a third faster.
+    * SlowScanner works extremely slowly but does not require machine determinization and
+      can be used to match complex expressions like /x.{50}$/ which cannot be compiled into Scanner.
+
+Required scanner type can be determined calling Fsm::Compile() or by explicitly constructing the desired scanner from Fsm.
+
+Encodings
+=========
+
+
+Pire supports Latin-1 and UTF-8 encodings by default. The required instances of the Pire::Encoding classes can be
+obtained by calling the appropriate functions from the namespace Pire::Encodings.
+
+Support for other encodings inheriting from Pire::Encoding can be added.
+In the inherited class, the following methods should be overridden:
+
+    * wchar32 FromLocal(const char*& begin, const char* end) reads and returns input character and forwards begin.
+      Returns an exceprion if the input includes an invalid sequence.
```

Comment:
Please, replace "exceprion" with "exception".

And it is more common to say "to throw an exception" than "to return an exception".